main.tf

```diff
@@ -1,8 +1,25 @@
-#################
-# Security group
-#################
+##########################
+# Security group with name
+##########################
 resource "aws_security_group" "this" {
-  count = "${var.create ? 1 : 0}"
+  count = "${var.create && var.use_name_prefix ? 0 : 1}"
+
+  name        = "${var.name}"
+  description = "${var.description}"
+  vpc_id      = "${var.vpc_id}"
+
+  tags = "${merge(var.tags, map("Name", format("%s", var.name)))}"
+
+  lifecycle {
```

Comment:
`create_before_destroy = true` will fail since `name` should be unique. This block should be removed from `this` security group.